### PR TITLE
[DNM][POAE7-1424] Add support for new Velox function via expr/translator/ircodegen

### PR DESCRIPTION
**Pull request by xieqi**

### CODE REVIEW

File: Analyzer/Analyzer.cpp [hunk 1]
```diff
@@ -1832,6 +1836,20 @@ void DatetruncExpr::group_predicates(std::list<const Expr*>& scan_predicates,
   }
 }
 
+void VeloxFunctionExpr::group_predicates(std::list<const Expr*>& scan_predicates,
```
Comment (winningsix) [inline Analyzer/Analyzer.cpp:1839]: Not related to this PR. Just interesting, where is group_predicates used?

File: QueryEngine/VeloxFunctionIR.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    throw QueryMustRunOnCpu();
+  }
+
+  const auto arg0 = expr->get_arg();
+  const auto& arg0_ti = arg0->get_type_info();
+  const auto arg0_lvs = codegen(arg0, true, co);
+  const auto arg0_lv = arg0_lvs.front();
+
+  std::string func_name = "";
+  if (arg0_ti.get_type() == kINT) {
```
Comment (winningsix) [inline QueryEngine/VeloxFunctionIR.cpp:42]: We need decouple this logic here. Frontend rule dispatcher needs this information as well. And those two places need to be consistent.

**Comment by winningsix** [comment]
A few follow ups needed:
1. This PR demonstrates a good path from analyzer to Velox function. why marked as DNM?
2. We need to separate supported function untils method. This will also be respected by plan dispatcher. @XinDongIntel
3. @yma11, this needs to touch RelAlgTranslator and also involves expressionRewrite. Do we need to keep expressionRewrite?

**Review by winningsix — CHANGES_REQUESTED** [review]

**Comment by yma11** [comment]
> A few follow ups needed:
> 
> 1. This PR demonstrates a good path from analyzer to Velox function. why marked as DNM?
> 2. We need to separate supported function untils method. This will also be respected by plan dispatcher. @XinDongIntel
> 3. @yma11, this needs to touch RelAlgTranslator and also involves expressionRewrite. Do we need to keep expressionRewrite?

My understanding is that RelAlgTranslator is required but expressionRewrite provided here is just a common method of Analyzer::Expr and we won't call it in our code path.

**Comment by xieqi** [comment]
> A few follow ups needed:
> 
> 1. This PR demonstrates a good path from analyzer to Velox function. why marked as DNM?
As the new function need calcite parser support currently the test will fail, so we can't test this path.

File: QueryEngine/VeloxFunctionIR.cpp (hunk 2 above)
Comment (xieqi) [reply QueryEngine/VeloxFunctionIR.cpp:42]: this is just a demo, you need customized this code according to your function parameters.

**Review by yma11 — APPROVED** [review]
LGTM.

**Comment by winningsix** [comment]
> > A few follow ups needed:
> > 
> > 1. This PR demonstrates a good path from analyzer to Velox function. why marked as DNM?
> >    As the new function need calcite parser support currently the test will fail, so we can't test this path.

So we suggest to close or refine this?

**Pull request closed without merging**